### PR TITLE
Storage fixes

### PR DESCRIPTION
**Pull request by LLParse**

### CODE REVIEW

File: docker/volumeplugin/plugin.go [hunk 1]
```diff
@@ -54,8 +53,8 @@ func NewRancherStorageDriver(driver string, client *client.RancherClient, cli *d
 	if err := d.init(); err != nil {
 		return nil, errors.Wrap(err, "Failed to initialize")
 	}
-	d.kickGC()
-	go d.watchContainerDeletes()
+	//d.kickGC()
```
Comment (cjellick) [inline docker/volumeplugin/plugin.go]: For the sake of keeping the code clean, can you just delete the lines you've commented out throughout this PR?

File: docker/volumeplugin/plugin.go [hunk 2]
```diff
@@ -79,70 +77,96 @@ func (d *RancherStorageDriver) init() error {
...
+
+	response := volume.Response{}
+	//defer logResponse("get", &response)
 
 	vol, _, err := d.state.Get(request.Name)
-	return volToResponse(err, vol)
+	if vol != nil {	
+		response.Volume = vol
+	}
+	if err != nil {
```
Comment (cjellick) [inline docker/volumeplugin/plugin.go:137]: I think you should check for an error first and _just_ return the error if it is not nil.

File: docker/volumeplugin/plugin.go [hunk 3]
```diff
@@ -79,70 +77,96 @@ func (d *RancherStorageDriver) init() error {
...
+	response := volume.Response{}
+	//defer logResponse("list", &response)
 
 	volumes, err := d.state.List()
 	if err != nil {
-		return volErr(err)
-	}
-	return volume.Response{
-		Volumes: volumes,
+		response.Err = err.Error()
```
Comment (cjellick) [inline docker/volumeplugin/plugin.go:122]: I think you should check for an error first and _just_ return the error if it is not nil.

File: docker/volumeplugin/plugin.go [hunk 4]
```diff
@@ -79,70 +77,96 @@ func (d *RancherStorageDriver) init() error {
...
 func (d *RancherStorageDriver) Create(request volume.Request) volume.Response {
-	logrus.Infof("Docker Create request: %v", request)
+	logRequest("create", &request)
+
+	response := volume.Response{}
+	defer logResponse("create", &response)
 
 	if created, err := d.state.IsCreated(request.Name); err != nil {
-		return volErr(err)
+		response.Err = err.Error()
```
Comment (cjellick) [inline docker/volumeplugin/plugin.go:88]: I think we should log non-nil errors at this point. This comment applies to all the top-level functions that docker calls,

File: package/common/common.sh [hunk 5]
```diff
@@ -98,19 +99,8 @@ ismounted() {
 }
 
 init_nfs_client_service() {
-    # using host network context to start rpcbind and rpc.statd inside the container process.
```
Comment (LLParse) [inline package/common/common.sh:101]: this needs to be put back for nfsv3

File: docker/volumeplugin/plugin.go (hunk 4 above)
Comment (LLParse) [reply docker/volumeplugin/plugin.go:88]: The pattern I've followed throughout is to set the error in the response and immediately return. At the top of each top-level function, I added a `defer logResponse(...)` that logs the response - but it always logs at info label. That should be changed to log at error level if an error is present in the response.

Comment (LLParse) [reply docker/volumeplugin/plugin.go:88]: Example with changes:

`time="2016-11-16T18:18:08Z" level=info msg=create.request name="elated-shannon_etcd_backup_a82d3" options=map[size:499 volumeType:st1]`

`time="2016-11-16T18:18:09Z" level=error msg=create.response error="Failed in create: \nAn error occurred (InvalidParameterValue) when calling the CreateVolume operation: The volume size is invalid for st1 volumes: 499 GiB. st1 volumes must be at least 500 GiB in size. Please specify a volume size above the minimum limit."`

**Comment by cjellick** [comment]
LGTM

**Pull request merged**